pilot/pkg/networking/core/v1alpha3/route/route.go

```diff
@@ -998,6 +1001,30 @@ func getEnvoyRouteTypeAndVal(r *route.Route) (envoyRouteType, string) {
 	return iType, iVal
 }
 
+func catchAllMatch(http *networking.HTTPRoute) *networking.HTTPMatchRequest {
+	// A Match is catch all if and only if it has no header/query param match
+	// and URI has a prefix / or regex *.
+	for _, match := range http.Match {
+		catchalluri := false
+		if match.Uri != nil {
```

Comment:
@ramaraochavali it is catch all too if uri = nil

Reply:
If uri is nil, but if it has headers/query params, we can not treat it as catchall right?

Reply:
yep. thats not catch all